test/fhirtypes/FHIRdateTime.test.ts

```diff
@@ -0,0 +1,51 @@
+import { FHIRdateTime, validateFHIRdateTime } from '../../src/fhirtypes';
+import { InvalidDateTimeError } from '../../src/errors/InvalidDateTimeError';
+
+describe('FHIRdateTime', () => {
+  let yearOnly: FHIRdateTime;
+  let yearMonth: FHIRdateTime;
+  let dateOnly: FHIRdateTime;
+  let hasTimeZone: FHIRdateTime;
+  let zTime: FHIRdateTime;
+  let noTimeZone: FHIRdateTime;
+
+  beforeAll(() => {
+    yearOnly = '1964';
+    yearMonth = '1975-05';
+    dateOnly = '2006-03-29';
+    hasTimeZone = '2010-02-07T13:28:17+03:00';
+    zTime = '2017-01-01T00:00:00.000Z';
+    noTimeZone = '2010-02-07T13:28:17';
```

Comment:
I think these tests would be cleaner if you just put the literal strings right into the test cases.  E.g.:
```ts
    it('should allow just a year', () => {
      validateFHIRdateTime('1964');
    });
```

Creating objects in the higher-level scope is really only useful if they're used in more than one test.